src/LatteContext/Collector/TemplatePathCollector/TemplatePathCollectorInterface.php

```diff
@@ -0,0 +1,18 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Efabrica\PHPStanLatte\LatteContext\Collector\TemplatePathCollector;
+
+use PhpParser\Node;
+use PHPStan\Analyser\Scope;
+
+interface TemplatePathCollectorInterface
+{
+    public function isSupported(Node $node): bool;
+
+    /**
+     * @return string[]|null
```

Comment:
Why string and not CollectedTemplatePath? In VariableSubCollectors we also return CollectedVariable. Would not be it better to keep it similar?

Reply:
Because you have there special case for null. If any collector find template in node, we have to return it. If CollectedTemplatePath is returned here, we have no chance to change it. So we would have errors which we don’t want.

Reply:
make sense. I jsut wonder if it would not be better for variable sub collectors also return just name=>type instead of collectedVariable

Reply:
Maybe, in my PR it was just simple add collected variable to array. Then it becames more complicated. I'm not sure what is better for that use case. But I don’t very like some array shaped returns (name, type). CollectedVariable seems nice to me.